packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -34,8 +34,8 @@ export class AwsCliCompatible {
       requestHandler: AwsCliCompatible.requestHandlerBuilder(options.httpOptions),
       customUserAgent: 'aws-cdk',
       logger: options.logger,
+      region: await this.region(options.profile),
```

Comment:
Is it possible to have unit test that ensures the `region` config is set so that the test will fail if one, in the future, accidentally removed this line?